fem/bilinearform.cpp

```diff
@@ -112,6 +113,138 @@ BilinearForm::BilinearForm (FiniteElementSpace * f, BilinearForm * bf, int ps)
    AllocMat();
 }
 
+BilinearForm::BilinearForm(BilinearForm &&other)
+   : Matrix(other.fes->GetVSize()), mat(other.mat), mat_e(other.mat_e),
+     fes(other.fes), assembly(other.assembly), batch(other.batch),
+     ext(other.ext), sequence(other.sequence), extern_bfs(other.extern_bfs),
+     element_matrices(other.element_matrices), static_cond(other.static_cond),
+     hybridization(other.hybridization), diag_policy(other.diag_policy),
+     precompute_sparsity(other.precompute_sparsity)
+{
+   // We swap stored integrators and markers with the moved nonlinear form
+   mfem::Swap(domain_integs, other.domain_integs);
+   mfem::Swap(domain_integs_marker, other.domain_integs_marker);
+   mfem::Swap(boundary_integs, other.boundary_integs);
+   mfem::Swap(boundary_integs_marker, other.boundary_integs_marker);
+   mfem::Swap(interior_face_integs, other.interior_face_integs);
+   mfem::Swap(boundary_face_integs, other.boundary_face_integs);
+   mfem::Swap(boundary_face_integs_marker, other.boundary_face_integs_marker);
```

Comment:
It is probably best to `DeleteAll()` these arrays in `other`, after deleting the owned pointers.

Actually, it's probably even better to define move ctor and/or move assignment for `Array` and use that here.

In any case we can factor out a method that deletes the owned pointers from the arrays and reuse it a a few places.

Reply:
Actually, in this method we don't need to delete any owned pointers -- that will be needed in the move assignment.

Reply:
Actually, in this method we don't need to delete any owned pointers -- that will be needed in the move assignment.